docs/application/native/guides/text-input/autofill.md

```diff
@@ -0,0 +1,365 @@
+    autofill_view_info_add_item(vi_h, ai_h);
+
+    if (app_id)
+        free(app_id);
+    ```
+
+3. Request to get the authentication information:
+
+    ```c
+    /* Send request to get authentication information */
+    ret = autofill_auth_info_request(ah, vi_h);
+    if (ret != AUTOFILL_ERROR_NONE)
+        /* Error handling */
+    ```
+
+4. When you no longer need the autofill view handle, call `autofill_view_info_destroy()` to release autofill view handle:
+
+    ```c
+    autofill_view_info_destroy(vi_h);
+    ```
+
+## Retrieving the data previously entered by the user and stored in an autofill service
+
+To Retrieve the data previously entered by the user and stored in an autofill service:
+
+1. Create the callback function to get the data previously entered by the user and stored in an autofill service:
+
+   Autofill service can send more than one autofill response for the view ID to be requested.
+   To support this case, `autofill_fill_response_foreach_group()` function is used to receive the multiple autofill response group. Each group has the autofill data for the requested view ID to be filled.
+
+   The following example code shows how to receive multiple autofill response data to have 2 autofill response to be consisted of user ID and user password:
+
+    ```c
+    static bool fill_response_item_cb(autofill_fill_response_item_h item, void *user_data)
+    {
+        char *id = NULL;
+        char *value = NULL;
+        char *presentation_text = NULL;
+
+        autofill_fill_response_item_get_id(item, &id);
+        autofill_fill_response_item_get_presentation_text(item, &presentation_text);
+        autofill_fill_response_item_get_value(item, &value);
+
+        dlog_print(DLOG_INFO, LOG_TAG, "id : %s, value : %s, presentation text : %s", id, value, presentation_text);
+
+        if (id) {
+            free(id);
+        }
+
+        if (value) {
+            free(value);
+        }
+
+        if (presentation_text) {
+            free(presentation_text);
+        }
+
+        return true;
+    }
+
+    static bool fill_response_group_cb(autofill_fill_response_group_h group_h, void *user_data)
+    {
+        autofill_fill_response_group_foreach_item(group_h, fill_response_item_cb, group_h);
+
+        return true;
+    }
+
+    static void
+    fill_response_received_cb(autofill_h ah, autofill_fill_response_h fill_response, void *data)
+    {
+        /* get fill response group count */
+        int count;
+        autofill_fill_response_get_group_count(fill_response, &count);
+        dlog_print(DLOG_INFO, LOG_TAG, "group count : %d", count);
+
+        autofill_fill_response_foreach_group(fill_response, fill_response_group_cb, &count);
+    }
+
+    /* Set callback function for receiving autofill fill response */
+    autofill_fill_response_set_received_cb(ah, fill_response_received_cb, NULL);
+    ```
+
+2. Create autofill data for each input field and view information to include several autofill data:
```

Comment:
To include several Autofill data, create Autofill data for each input field and view information:

Reply:
fixed.